lib/api/objectPut.js

```diff
@@ -78,18 +81,29 @@ function _storeIt(bucketName, objectKey, objMD, authInfo, canonicalID,
     // 0 or > 0 - existing object with content-length 0 or greater than 0
     const prevContentLen = objMD && objMD['content-length'] !== undefined ?
         objMD['content-length'] : null;
+
+    const backendInfoObj = locationConstraintCheck(request, null, bucket, log);
+    if (backendInfoObj.err) {
+        return process.nextTick(() => {
```

Comment:
let's get rid of it here though since the function is already returning an err above directly.  at some point, we could do a clean up pr to make this all consistent.